source/tit/sph/particle_mesh.hpp

```diff
@@ -156,13 +142,12 @@ class ParticleMesh final {
   }
 
   /// Adjacent fixed particles.
-  template<class ParticleArray>
-  constexpr auto operator[](std::nullptr_t,
-                            ParticleArray& particles,
-                            size_t i) const noexcept {
+  template<class PV>
+  constexpr auto fixed_interp(PV a) const noexcept {
+    const auto i = a.index() - (*a.array().fixed().begin()).index();
```

Comment:
Let's add an assert for the type of `a` and introduce a difference operator for the particle views.